acnportal/acnsim/network/current.py

```diff
@@ -6,27 +9,47 @@ class Current(pd.Series):
     Includes addition, subtraction, and multiplication (by scalar) operators.
 
     Attributes:
-        loads (Dict[str, number]): Dictionary which maps a load_id to its coefficient in the aggregate current.
+        loads (Dict[str, number]): Dictionary which maps a load_id to its coefficient in
+        the aggregate current.
 
     Args:
-        loads (Dict[str, number], str, or List[str]): If dict, a dictionary mapping load_ids to coefficients. If str a
-            load_id. If list, a list of load_ids. Default None. If None, loads will begin as an empty dict.
+        loads (Dict[str, number], str, or List[str], pd.Series): If dict, a dictionary
+            mapping load_ids to coefficients. If str a load_id. If list, a list of
+            load_ids. Default None. If None, loads will begin as an empty dict.
     """
 
-    def __init__(self, loads=None):
+    def __init__(
+        self,
+        loads: Union[Dict[str, SupportsFloat], str, List[str], pd.Series] = None,
+        **kwargs
```

Comment:
I think passing up kwargs to the Series constructor makes sense, as this lets the user specify some fields (like name and dtype) of the Series directly through the current constructor. The only wrinkle is what to do with the data kwarg, which is normally used to input the data to a Series. I just ignore it here in favor of the loads kwarg. Let me know what you think.

Reply:
Hmm. Seems ok, but the constructor is now quite complicated. Read we need to support lots of different inputs moving forward. It is possible that these would not be compatible with how currents are supposed to be used. 
---
This may not be the place for this, but I am starting to wonder why Current is a subclass of pd.Series at all. I seems it could either:
1) Contain a Series object
2) Be a factory method which returns a series object.

Reply:
If I recall, I had originally written Current as a subclass of Series for two reasons:

1. This gives us access to a lot of Series methods that we wouldn't have access to if we used an object consisting of a Series, at least not directly. E.g. DataFrame appending, reindexing, scalar multiplication, etc. A lot of these methods are used by ChargingNetwork to build the constraint matrix. If we use composition instead of inheritance, we'll have to write passing methods to the internal Series attribute.

2. Subclassing Series rather than using Series directly allows us to override certain specific methods of Series, basically the add and subtract methods. If we used Series directly, we couldn't use + and - operators alone (we'd have to fillna(0) or something like that)

I agree that we can probably not allow Series kwargs to the Current constructor and no one would complain, so I can revert those changes. As for whether or not we should inherit from  Series in the first place, I think that warrants a discussion in a different PR.